src/pyhf/infer/intervals.py

```diff
@@ -54,6 +169,19 @@ def upperlimit(data, model, scan, level=0.05, return_results=False):
               :class:`~pyhf.infer.hypotest` results at each test POI.
               Only returned when ``return_results`` is ``True``.
     """
+
+    if isinstance(scan, str):
+        if scan.lower() == 'auto':
+            bounds = model.config.suggested_bounds()[
+                model.config.par_slice(model.config.poi_name).start
+            ]
+            obs_limit, exp_limit, results = upperlimit_auto(
+                data, model, bounds[0], bounds[1], rtol=1e-3, from_upperlimit_fn=True
+            )
+            if return_results:
+                return obs_limit, exp_limit, results
+            return obs_limit, exp_limit
+
```

Comment:
thanks a lot @beojan -- could I ask you to factor the below out into `upperlimit_fixedscan` and then we have

```
    if isinstance(scan, str) and scan.lower() == 'auto':
            obs_limit, exp_limit, results = upperlimit_auto(
                data, model, bounds[0], bounds[1], rtol=1e-3, from_upperlimit_fn=True
            )
   else:
            obs_limit, exp_limit, results = upperlimit_scan(
                data, model, scan
            )            
            ...
```


to get   a bit more symmetry ?

Reply:
No worries. We (or at least I) need to get back to this and give it another look over and review. I was hoping to figure out a way to have improved our release setup to make feature contributions like yours easier to fit in with patch vs. minor release cycles, but I don't think it is worth it here and we should just get your contribution in once it is ready and reviewed. :+1: I don't want to make you wait a huge amount of time (more than the time you've already waited).